src/main/java/com/crudapp/crudapp/services/CourseServiceImpl.java

```diff
@@ -90,10 +85,21 @@ public List<Object[]> getUsersAndCoursesByCourseId(Long courseId) {
 
      }
 
-//     @Override
-//     public List<Course> findByDescription(String description) {
-//         Course one=courseDao.findbyDescrition(description);
-//         return null;
-//     }
+     public UserCourseDto getUserAndCourseByUserId(Long Id)
+     {
+         Long courseId= userRepo.FindCourseIdbyUserId(Id);
+
+         if(courseId!=null)
+         {
+             Course course=courseDao.FindCourseById(courseId);
```

Comment:
Use best practices: camelCasing naming convention should be used for functions.
```suggestion
             Course course=courseDao.findCourseById(courseId);
```